waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/client/DefaultMetaStoreClientFactory.java

```diff
@@ -43,7 +43,7 @@ private static class ReconnectingMetastoreClientInvocationHandler implements Inv
     private final String name;
     private final int maxRetries;
 
-    private HiveUgiArgs cachedUgi = HiveUgiArgs.WAGGLE_DANCE_DEFAULT;
+    private HiveUgiArgs cachedUgi = null;
```

Comment:
removed the default to be more compatible a client would behave if setUgi is never called before doing a real call. We might be seeing this in Trino clients.